test/src/problems/FixedPointProblem.C

```diff
@@ -0,0 +1,58 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FixedPointProblem.h"
+
+#include "NonlinearSystemBase.h"
+
+registerMooseObject("MooseTestApp", FixedPointProblem);
+
+template <>
+InputParameters
+validParams<FixedPointProblem>()
+{
+  InputParameters params = validParams<FEProblem>();
+  params.addParam<TagName>("fp_tag_name", "fp_previous", "Tag name for the fixed point iteration");
+  return params;
+}
+
+FixedPointProblem::FixedPointProblem(const InputParameters & params)
+  : FEProblem(params),
+    _tag_previous(getParam<TagName>("fp_tag_name")),
+    _tag_id(addVectorTag(_tag_previous)),
+    _residual_previous(_nl->addVector(_tag_id, false, GHOSTED))
+{
+}
+
+void
+FixedPointProblem::computeResidual(const NumericVector<Number> & soln,
+                                   NumericVector<Number> & residual)
+{
+  // excluding the previous tag evaluation
+  auto & tags = getVectorTags();
+
+  _fe_vector_tags.clear();
+
+  for (auto & tag : tags)
+    if (tag.second != _tag_id)
+      _fe_vector_tags.insert(tag.second);
+
+  computeResidualInternal(soln, residual, _fe_vector_tags);
+
+  residual += _residual_previous;
+}
+
+void
+FixedPointProblem::computeFullResidual(const NumericVector<Number> & soln,
+                                       NumericVector<Number> & residual)
+{
+  _residual_previous.close();
```

Comment:
Otherwise I got an assertion error in debug mode. Initially I thought I do not need this. I am lazy to figure out where the close is missing. On the other hand, if `close` has a quick return based on `closed()`, the performance is not hurt.

Reply:
There is no way that  `close` can quickly return without a global communication.  It is definitely impossible. 

Suppose there are three mpi processes:

mpi 0: closed_flag: true

mpi 1: closed_flag: false (triggered by any `MatSetValues`)

mpi 2: closed_flag: true

Only everybody has true, and we can not return. How `mpi` 0 an `mpi` 2 know the flag should be false without the communication.

Reply:
Although I am not fully convinced about that closing a closed vector is costly, I agree that we should remove unnecessary close calls. This call is immediately after `FEProblem::solve`, and we would think all vectors have been properly closed. But it is not here. It is pretty hard to figure out what is missing in the `solve`.

Reply:
I will spend a little more time on why the tagged vector is not closed properly.